lib/screens/signup/signup_repository.dart

```diff
@@ -1,15 +1,21 @@
 import 'dart:math';
 
 import 'package:ecellapp/core/res/errors.dart';
+import 'package:ecellapp/core/res/strings.dart';
+import 'package:ecellapp/core/utils/injection.dart';
+import 'package:ecellapp/core/utils/logger.dart';
+import 'package:http/http.dart' as http;
 
 abstract class SignupRepository {
-  /// Takes in `name`, `email`, `password` and `mobileNumber` and returns a success code OR a suitable exception
-  Future<void> signup(String name, String email, String password, int mobileNumber);
+  /// Takes in `firstName`, `lastName`, `email`, `mobileNumber` and `password` and returns a success code OR a suitable exception
```

Comment:
🟤 
Return type is Future<void>, it doesn't return a success code.
```suggestion
  /// Takes in `firstName`, `lastName`, `email`, `mobileNumber` and `password`, registers the user and throws a suitable exception if something goes wrong.
```

Reply:
done